src/Database/Type/DateTimeType.php

```diff
@@ -103,7 +103,7 @@ public function toDatabase($value, Driver $driver)
      */
     public function toPHP($value, Driver $driver)
     {
-        if ($value === null) {
+        if ($value === null || (int)$value === 0) {
```

Comment:
Well, that date is not `null` (as in null worthy) per se, so I wouldn't want to catch/support that either way.